### PR TITLE
Taskgraph follow-up fixes

### PR DESCRIPTION
**Pull request by bhearsum**
* Raise level to 3 to match ci-config; build for master/dev/production.
* Fix secret parsing
* Create archive tag in addition to branch tag
* We need the version to create the archive tag properly
* Typo (GIT_HEAD_REV vs VCS_HEAD_REV)

### CODE REVIEW

**Comment by bhearsum** [comment]
taskcluster-github complaints are https://github.com/taskcluster/taskcluster/issues/6984, not indicative of a real problem.

File: .taskcluster.yml [hunk 1]
```diff
@@ -96,7 +100,7 @@ tasks:
           in:
               $if: >
                   tasks_for in ["action", "pr-action", "cron"]
-                  || (tasks_for == "github-push" && short_head_ref == "main")
+                  || (tasks_for == "github-push" && (short_head_ref == "master" || short_head_ref == "dev" || short_head_ref == "production"))
```
Comment (ahal) [inline .taskcluster.yml:103]: nit: You can use `in` here

**Review by ahal — APPROVED** [review]
Thanks, lgtm!

File: .taskcluster.yml (hunk 1 above)
Comment (bhearsum) [reply .taskcluster.yml:103]: Yeah, good call.

**Pull request merged**